core/src/main/java/org/infinispan/configuration/cache/AbstractStoreConfiguration.java

```diff
@@ -9,8 +9,8 @@
 import org.infinispan.commons.util.TypedProperties;
 
 public class AbstractStoreConfiguration implements StoreConfiguration {
-   public static final AttributeDefinition<Boolean> FETCH_PERSISTENT_STATE = AttributeDefinition.builder("fetchPersistentState", false).immutable().build();
-   public static final AttributeDefinition<Boolean> PURGE_ON_STARTUP = AttributeDefinition.builder("purgeOnStartup", false).immutable().build();
+   public static final AttributeDefinition<Boolean> FETCH_PERSISTENT_STATE = AttributeDefinition.builder("fetchPersistentState", false).xmlName("").immutable().build();
+   public static final AttributeDefinition<Boolean> PURGE_ON_STARTUP = AttributeDefinition.builder("purge", false).immutable().build();
```

Comment:
IMO, it would be better to leave the attribute name unchanged and use ```xmlName("purge")```. In most cases, the attribute name is more specific than the xml name (another example, ```BackupConfigure.REPLICATION_TIMEOUT```)